include/igl/copyleft/comiso/miq.cpp

```diff
@@ -465,28 +469,29 @@ IGL_INLINE std::vector<std::vector<typename igl::copyleft::comiso::VertexIndexin
   {
     isStartVertex[i] = false;
     // vertices with two neighbors are regular vertices, unless the vertex is a singularity, in which case it qualifies as a start vertex
-    if (VVSeam[i].size() > 0 && VVSeam[i].size() != 2 || singular(i) == true)
+    if ((!VVSeam[i].empty() && VVSeam[i].size() != 2) || singular(i) != 0)
     {
       startVertexIndices.push_back(i);
       isStartVertex[i] = true;
     }
   }
 
   // For each startVertex, walk along its seam
-  for (unsigned int i=0;i<startVertexIndices.size();i++)
+  for (auto element : startVertexIndices)
   {
-    auto startVertexNeighbors = &VVSeam[startVertexIndices[i]];
-    const int neighborSize = startVertexNeighbors->size();
+    auto startVertexNeighbors = &VVSeam[element];
+    size_t neighborSize = startVertexNeighbors->size();
 
     // explore every seam to which this vertex is a start vertex
     // note: a vertex can never be a start vertex and a regular vertex simultaneously
-    for (unsigned int j=0;j<neighborSize;j++)
+    for (size_t j = 0; j < neighborSize; j++)
     {
       std::vector<VertexInfo> thisSeam; // temporary container
 
       // Create vertexInfo struct for start vertex
-      auto startVertex = VertexInfo(startVertexIndices[i], -1, -1, -1, -1);// -1 values are arbitrary (will never be used)
-      auto currentVertex = startVertex;
+      VertexInfo startVertex = VertexInfo(element, -1, -1, -1, -1);// -1 values are arbitrary (will never be used)
+      VertexInfo currentVertex = startVertex;
+      VertexInfo prevVertex = startVertex;
```

Comment:
Can you leave this one before the `while` and keep the comment?

Reply:
ok